### PR TITLE
Add GCP stackdriver aggregation parameters

### LINKED ISSUE #3008: Introduce metric aggregation/alignment in GCP Stackdriver scaler
### Proposal

It's crucial to have ability applying aggregation/alignment on GCP Stackdriver metric in order to both fetch less data from GCP API and get exactly what you need.

For instance, `aggregator` and `aligner` would be useful:

```yaml
triggers:
- type: gcp-stackdriver
  metadata:
    projectId: my-project-id
    filter: 'metric.type="loadbalancing.googleapis.com/https/backend_latencies" AND resource.type="https_lb_rule" AND resource.label.forwarding_rule_name="k8s2-..."'
    aggregator: 'mean | sum | 95th_percentile | etc.'
    aligner: 'delta | sum | 95th_percentile | etc.'
    targetValue: '10' # latency in ms
    credentialsFromEnv: GOOGLE_APPLICATION_CREDENTIALS_JSON
```

### Use-Case

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by RamCohen**
Fixes #3008

Docs: https://github.com/kedacore/keda-docs/pull/776

Signed-off-by: Ram Cohen <ram.cohen@gmail.com>

### CODE REVIEW

**Comment by RamCohen** [comment]
Will open a documentation PR if this is tentatively approved

File: pkg/scalers/gcp_stackdriver_scaler.go [hunk 1]
```diff
@@ -5,14 +5,14 @@ import (
 	"fmt"
 	"strconv"
 
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
```
Comment (zroubalik) [inline pkg/scalers/gcp_stackdriver_scaler.go:8]: same as above

File: pkg/scalers/gcp_pubsub_scaler.go [hunk 2]
```diff
@@ -60,6 +61,11 @@ func NewPubSubScaler(config *ScalerConfig) (Scaler, error) {
 		return nil, fmt.Errorf("error parsing PubSub metadata: %s", err)
 	}
 
+	meta.aggregation, err = NewStackdriverAggregator(60, "mean", "none")
```
Comment (zroubalik) [inline pkg/scalers/gcp_pubsub_scaler.go:64]: Just to clarify for PubSub it doesn't make sense to support this feature, right?

**Review by zroubalik — COMMENTED** [review]
Looking good in general, nice impovement, thanks!

**Comment by RamCohen** [comment]
Not sure what to do with the annoying unrelated static check failure:
pkg/scalers/azure_blob_scaler.go:131:102: string `true` has 3 occurrences, make it a constant (goconst)

**Comment by JorTurFer** [comment]
> Not sure what to do with the annoying unrelated static check failure:
> pkg/scalers/azure_blob_scaler.go:131:102: string `true` has 3 occurrences, make it a constant (goconst)

IDK why this happens only in your PR and not in others... You could just refactor it a bit, something like this:
```golang
	if val, ok := config.TriggerMetadata["useAAdPodIdentity"]; ok {
		value, err := strconv.ParseBool(val)
		if err != nil {
			return nil, "", err
		}
		if config.PodIdentity == "" && value {
			config.PodIdentity = kedav1alpha1.PodIdentityProviderAzure
		}
	}
```

**Comment by zroubalik** [comment]
> > Not sure what to do with the annoying unrelated static check failure:
> > pkg/scalers/azure_blob_scaler.go:131:102: string `true` has 3 occurrences, make it a constant (goconst)
> 
> IDK why this happens only in your PR and not in others... You could just refactor it a bit, something like this:
> 
> ```go
> 	if val, ok := config.TriggerMetadata["useAAdPodIdentity"]; ok {
> 		value, err := strconv.ParseBool(val)
> 		if err != nil {
> 			return nil, "", err
> 		}
> 		if config.PodIdentity == "" && value {
> 			config.PodIdentity = kedav1alpha1.PodIdentityProviderAzure
> 		}
> 	}
> ```

It is happening because another `"true"` has been added in this PR in pkg/scalers/stackdriver_client.go.

**Comment by JorTurFer** [comment]
> It is happening because another `"true"` has been added in this PR in pkg/scalers/stackdriver_client.go.

lol, good catch, I didn't see it 👁️

**Comment by RamCohen** [comment]
> 

changed the switch statement to use "count_true" and "count_false" so that should solve it

**Review by zroubalik — APPROVED** [review]
LGTM

**Comment by zroubalik** [comment]
@RamCohen e2e tests failed on this one. Btw we also have started a migration of e2e tests from TS to Go, do you think you can try to tackle this? Thanks!

**Comment by RamCohen** [comment]
I saw that but I can't see how to get to the logs to see what happens

As for TS to Go: Will be happy to once I see some examples of other tests

**Comment by JorTurFer** [comment]
> I saw that but I can't see how to get to the logs to see what happens

Hey,
You can click in the link that the GH action has updated and scroll inside the step "Run end to end tests". This is the link to [one of the logs](https://github.com/kedacore/keda/runs/6878375998?check_suite_focus=true#step:6:138)
image

**Comment by zroubalik** [comment]
@JorTurFer mind looking at this^

**Comment by zroubalik** [comment]
@RamCohen could you please update Changelog?

**Comment by ritmas** [comment]
When could this be released @zroubalik @RamCohen ?

**Pull request merged**